### PR TITLE
Replace Pino with Winston Logger - Part 3/4 

### PR DESCRIPTION
**Pull request by yashmurty**
Add `logWithContext` function. This will be renamed to `log` once we replace it across all controllers, services, etc.
Will replace this in all modules in `PR 4/4`.

### CODE REVIEW

**Review by sivanpayyadakath — APPROVED** [review]
LGTM

**Review by war1oc — APPROVED** [review]
LGTM

File: src/app.controller.spec.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { RequestContext } from './shared/request-context/request-context.dto';
 
 describe('AppController', () => {
   let moduleRef: TestingModule;
-  const mockedLogger = { setContext: jest.fn(), log: jest.fn() };
+  const mockedLogger = { setContext: jest.fn(), logWithContext: jest.fn() };
```
Comment (war1oc) [inline src/app.controller.spec.ts:9]: I think it's better to initialize these in beforeEach so that the mocks are clean for each test.

Comment (yashmurty) [reply src/app.controller.spec.ts:9]: Alright, let's change it across all in a separate PR. 🙏

**Pull request merged**